src/compiler/sys.ts

```diff
@@ -335,7 +335,12 @@ namespace ts {
                     data = "\uFEFF" + data;
                 }
 
-                _fs.writeFileSync(fileName, data, "utf8");
+                if (fileName === "/dev/stdout" || fileName === "-") {
+                    process.stdout.write(data, "utf8");
+                }
+                else {
+                    _fs.writeFileSync(fileName, data, "utf8");

```

Comment:
@mhegazy @mklement0 I tried to analyze why node returns `ESPIPE, invalid seek` so I cloned node repo from Github, compiled it and run:

```
$ echo "require('fs').writeFileSync('/dev/stdout', 'hello');" | ./node 

fs.js:707
    return binding.writeBuffer(fd, buffer, offset, length, position);
                   ^

Error: ESPIPE: invalid seek, write
    at Error (native)
    at Object.fs.writeSync (fs.js:707:20)
    at Object.fs.writeFileSync (fs.js:1235:24)
    at [stdin]:1:15
    at Object.exports.runInThisContext (vm.js:54:17)
    at Object.<anonymous> ([stdin]-wrapper:6:22)
    at Module._compile (module.js:425:26)
    at node.js:589:27
    at doNTCallback0 (node.js:430:9)
    at process._tickCallback (node.js:359:13)
```

So we have [the line](https://github.com/nodejs/node/blob/master/lib/fs.js#L706) with `binding.writeBuffer` where parameter values are `11 <Buffer 68 65 6c 6c 6f> 0 5 0` (I simply used `console.log()` to output parameter values).

`binding.writeBuffer` call leads to [calling](https://github.com/nodejs/node/blob/master/src/node_file.cc#L907) `SYNC_CALL(write, nullptr, fd, &uvbuf, 1, pos)` and this command throws an exception (a debug statement of mine after this call is not printed). `SYNC_CALL(write, ...)` leads to calling `uv_fs_write` [function](https://github.com/nodejs/node/blob/master/deps/uv/src/unix/fs.c#L1209) and it finishes, I don't know where exactly is the exception thrown (it's not in [`SYNC_DEST_CALL`](https://github.com/nodejs/node/blob/master/src/node_file.cc#L292)).

To make a conclusion, the problem seems to be in the fact that `fs.writeFileSync` uses a buffer implementation which sets offset value `0` for writing to a file (it is stdout in our case). However, this is inconclusive because I don't know Node architecture and my knowledge of C++ is fairly basic.